src/App.js

```diff
@@ -1,16 +1,118 @@
 import React from 'react';
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+import Todo from './components/TodoComponents/Todo.js'
+
+import './App.css'
+
+const todoListData = [
+  {
+    task: 'Learn User Interface and Git',
+    id: 1,
+    completed: false
+  },
+  {
+    task: 'Learn Advanced CSS',
+    id: 2,
+    completed: false
+  },
+  {
+    task: 'Learn JavaScript Fundamentals',
+    id: 3,
+    completed: false
+  },
+  {
+    task: 'Learn Applied JavaScript',
+    id: 4,
+    completed: false
+  },
+  {
+    task: 'Complete Project Week: User Interface',
+    id: 5,
+    completed: false
+  },
+  {
+    task: 'Learn React',
+    id: 6,
+    completed: false
+  },
+  {
+    task: 'Learn Intermediate React',
+    id: 7,
+    completed: false
+  }
+]
+
+let lastId = todoListData[todoListData.length - 1].id;
+//used to keep track of new item Ids
+const addNewId = () => {
+  lastId++;
+  return lastId;
+};
+
+//console.log(lastId);
+
+
 
 class App extends React.Component {
-  // you will need a place to store your state in this component.
-  // design `App` to be the parent component of your application.
-  // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+
+    this.state = {
+      todoListData,
+    };
+  };
+  
+  addItem = (e, item) => {
+    e.preventDefault();
+    const newItem = {
+      task: item,
+      id: addNewId(),
+      completed: false
+    };
+    this.setState({
+      todoListData: [...this.state.todoListData, newItem]
+    });
+  };
+
```

Comment:
all your functionality seems to work when I test it so great job on learning how to use these handler functions/methods! I saw in your commits it took a bit of debugging so great job sticking to the problem and over coming it